agent/consul/server.go

```diff
@@ -1646,6 +1645,7 @@ func (s *Server) ReloadConfig(config ReloadableConfig) error {
 	s.rpcConnLimiter.SetConfig(connlimit.Config{
 		MaxConnsPerClientIP: config.RPCMaxConnsPerClient,
 	})
+	s.connPool.SetRPCClientTimeout(config.RPCClientTimeout)
```

Comment:
Happy to leave it for now if there is time pressure to merge, but I didn't see a reload test covering this like you added in client_test.go. There might even be an existing Reload test you can just add a couple of lines to to cover this. Fine to do it as a follow up though!

Reply:
Added!